### PR TITLE
Fix soecp evaluation

### PR DESCRIPTION
**Pull request by camelto2**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

This fixes a bug in the evaluation of the SOECPComponents. Essentially there was a mismatch in the normalization of integrals involved as well as a swapped conjugation of spherical harmonics. When I implemented the reference evaluation, I made the same mistake. I noticed this when comparing against my QWalk implementation and some calculations from DIRAC. This fix is reflected in the evaluation and unit tests. 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
my workstation, gcc 9.2 
## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by rcclay** [comment]
OK to test.

**Review by rcclay — APPROVED** [review]

**Comment by camelto2** [comment]
The failing alcf-clang-CUDA tests is in some AFQMC stuff only in the mixed precision version...which I obviously did not touch any of that code.

**Comment by ye-luo** [comment]
The failure is not from AFQMC. Search for error.
```
/sandbox/jenkins-ci/workspace/QMCPACK-CI-Clang-CUDA/label/skylake-only/type/Complex-Mixed-enable-CUDA/src/QMCHamiltonians/SOECPComponent.cpp:126:60: error: invalid operands to binary expression ('complex<float>' and 'double')
    psiratio[j] = Psi.calcRatio(W, iel) * sgridweight_m[j] * 2.0 * TWOPI;
```

**Comment by camelto2** [comment]
> The failure is not from AFQMC. Search for error.
> 
> ```
> /sandbox/jenkins-ci/workspace/QMCPACK-CI-Clang-CUDA/label/skylake-only/type/Complex-Mixed-enable-CUDA/src/QMCHamiltonians/SOECPComponent.cpp:126:60: error: invalid operands to binary expression ('complex<float>' and 'double')
>     psiratio[j] = Psi.calcRatio(W, iel) * sgridweight_m[j] * 2.0 * TWOPI;
> ```

Just noticed that in one of the other failures. Fixing now

File: src/QMCHamiltonians/SOECPComponent.cpp [hunk 1]
```diff
@@ -119,11 +119,12 @@ SOECPComponent::ComplexType SOECPComponent::getAngularIntegral(RealType sold,
                                                                const PosType& dr)
 {
   //quadrature sum for angular integral
+  RealType fourpi = 2.0 * TWOPI;
```
Comment (ye-luo) [inline src/QMCHamiltonians/SOECPComponent.cpp:122]: constexpr should be better.

**Review by ye-luo — APPROVED** [review]

**Pull request merged**